components/proxy/src/main/java/com/hotels/styx/routing/handlers/PathPrefixRouter.java

```diff
@@ -47,42 +43,61 @@
 import static com.hotels.styx.routing.config.RoutingConfigParser.toRoutingConfigNode;
 import static com.hotels.styx.routing.config.RoutingSupport.missingAttributeError;
 import static java.lang.String.join;
-import static java.util.Comparator.comparingInt;
-import static java.util.Comparator.naturalOrder;
-import static java.util.concurrent.CompletableFuture.completedFuture;
-import static java.util.stream.Collectors.toList;
+import static java.util.Collections.singletonList;
+import static java.util.Objects.requireNonNull;
 
 /**
  * Makes a routing decision based on a request path prefix.
- *
+ * <p>
  * Chooses a destination according to longest matching path prefix.
  * The destination can be a routing object reference or an inline definition.
  */
-public class PathPrefixRouter {
-    public static final Schema.FieldType SCHEMA = object(
-            field("routes", list(object(
-                    field("prefix", string()),
-                    field("destination", routingObject()))
-            ))
-    );
-
-    private final ConcurrentSkipListMap<String, RoutingObject> routes = new ConcurrentSkipListMap<>(
-                    comparingInt(String::length)
-                            .reversed()
-                            .thenComparing(naturalOrder())
-    );
+public class PathPrefixRouter implements RoutingObject {
```

Comment:
The underlying routing object shouldn't be public. Because they are always created via the Factory class. You could make it a package private for the unit tests.

Reply:
The constructor is package private so instances of this class cannot be created (outside of the package) except via the factory.

Reply:
Ok. Lets explore some options as a future enhancement.